File: scala/src/main/scala/BankActor.scala
import akka.actor.{Props, ActorRef, Actor}

class BankActor extends Actor {
  var accounts = Map[String, ActorRef]()

  def receive = {
    case Open(accountId) => {
      accounts += (accountId -> context.actorOf(Props[BankAccountActor]))
    }

    case Transfer(fromId, toId, amount) => {
      accounts.get(fromId).get ! Withdrawal(fromId, amount)
      accounts.get(toId).get ! Deposit(toId, amount)
    }

    case msg: BankMessage => {
      accounts.get(msg.accountId).get forward msg
    }
  }
}

File: csharp/AkkaBank/BankActor.cs
using System.Collections.Generic;
using Akka.Actor;

namespace AkkaBank
{
    public class BankActor : ReceiveActor
    {
        private readonly IDictionary<string, IActorRef> accounts = new Dictionary<string, IActorRef>();

        public BankActor()
        {
            Receive<Open>(msg =>
            {
                accounts.Add(msg.AccountId, Context.ActorOf<BankAccountActor>());
            });

            Receive<Transfer>(msg =>
            {
                accounts[msg.AccountId].Tell(new Withdrawal {Amount = msg.Amount});
                accounts[msg.ToId].Tell(new Deposit {Amount = msg.Amount});
            });

            Receive<BankMessage>(msg => accounts[msg.AccountId].Forward(msg));
        }
    }
}